actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -437,6 +442,9 @@ def link_to_if(condition, name, options = {}, html_options = {}, &block)
       # Creates a mailto link tag to the specified +email_address+, which is
       # also used as the name of the link unless +name+ is specified. Additional
       # HTML attributes for the link can be passed in +html_options+.
+      # Note the order of params, which is reversed compared to +link_to+. Provide
+      # the part that is used for the href first, and the part that is visible
+      # to users second.
```

Comment:
> I don't agree that the examples are a replacement for the text documentation. Following this logic there is no need to write text at all.

I'm not sure how you got to there. For a start, the `link_to` method's text description doesn't state the order of the arguments. Let's not pretend the method signature isn't part of the documentation.

---

"Note" is way too intense, and the verbose description of the order is redundant with the signature. 

We could try something like "In contrast with `link_to`, the email address is passed before the link text." -- but I'm hesitant to add anything at all... the people making this mistake aren't going to be closely reading the documentation first. And it doesn't particularly matter anyway, because the cause (and thus solution) will be obvious when they get a link that says `<a href="mailto:email me">bob@example.com</a>`.